src/uri/uri.cc

```diff
@@ -249,6 +249,54 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  auto is_relative = path.starts_with("./") || path.starts_with("../");
+  if (is_relative) {
+    throw URIError{
+        "You cannot set a relative path, use URI::resolve_from() instead"};
+  }
+
+  auto parsed_path = URI{std::string{path}};
+  if (!parsed_path.path().has_value()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const bool has_leading_slash = parsed_path.path().value().front() == '/';
+  this->path_ = has_leading_slash ? parsed_path.path().value().substr(1)
```

Comment:
Why do we need this logic? Isn't this enough?

```suggestion
  this->path_ = parsed_path.path();
```

Reply:
Why?

Reply:
let check, but there is an isse with tests where need it! at some point out of the getter, but we decided to not keep it in the `path_` itself!

Reply:
I think you should take a proper look at that. The getter and setter should mirror each other, so a developer would expect that anything you "get" can be "set" without additional modifications, otherwise the setter is doing more work than it should?

Reply:
Yeah I agree! I'll take a deeper look tomorrow :)